geopandas/geodataframe.py

```diff
@@ -375,18 +375,17 @@ def from_features(cls, features, crs=None, columns=None):
             features_lst = features
 
         rows = []
-        for f in features_lst:
-            if hasattr(f, "__geo_interface__"):
-                f = f.__geo_interface__
-            else:
-                f = f
-
-            d = {"geometry": shape(f["geometry"]) if f["geometry"] else None}
-            d.update(f["properties"])
-            rows.append(d)
-        df = GeoDataFrame(rows, columns=columns)
-        df.crs = crs
-        return df
+        for feature in features_lst:
+            # load geometry
+            if hasattr(feature, "__geo_interface__"):
+                feature = feature.__geo_interface__
+            row = {
+                "geometry": shape(feature["geometry"]) if feature["geometry"] else None
+            }
+            # load properties
+            row.update(feature["properties"])
+            rows.append(row)
+        return GeoDataFrame(rows, columns=columns, crs=crs)
```

Comment:
I have the similar request as last the time. This is essentially the same piece of code, just using different variable name. Can you limit the diff again? Since you changed it to have the actual functionality in `read_file`, `from_features` could be kept intact within this PR. Otherwise it is perfect.

Reply:
Just updated to keep the PR more focused.

Reply:
Thank you. Once green we can merge.